### PR TITLE
Add hearing summary to prosecution case

### PR DESCRIPTION
**Pull request by mpw5**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/-CACP88)

Hearing summaries are part of the search response returned by Common Platform. This PR adds a `HearingSummary` model and serializer, and includes it in the existing `prosecution_case` model.

## Checklist

Before you ask people to review this PR:

- [x] Tests and linters should be passing
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/models/hearing_summary.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+class HearingSummary
+  include ActiveModel::Model
+
+  attr_accessor :body, :hearing_date
```
Comment (archferns) [inline app/models/hearing_summary.rb]: I think we are missing where we pass `hearing_date` to the model as well

File: app/models/defendant.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 class Defendant
   include ActiveModel::Model
 
-  attr_accessor :body
+  attr_accessor :body, :gender, :address_1, :address_2, :address_3, :address_4, :address_5, :postcode
```
Comment (archferns) [inline app/models/defendant.rb]: I think we are missing the initilization here (apart from the `body`)

Comment (mpw5) [reply app/models/defendant.rb]: `:gender, :address_1, :address_2, :address_3, :address_4, :address_5, :postcode` aren't actually part of the defendant data we get from the prosecution_case search results, so they won't appear in the body. That's the data we need to get by doing subsequent calls to get additional hearing data. 

My original thought was that we could update the `body` with that data after doing those calls, but after thinking about it it seemed easier to hold them as separate attributes on the model, hence this change.

You're right that they don't get initialised here, they would only get set later on, after we've done those extra hearing calls.

File: app/models/hearing_summary.rb (hunk 1 above)
Comment (mpw5) [reply app/models/hearing_summary.rb]: Similar to the `defendant` model, the `hearing_date` isn't data we receive in the `body` of a hearing summary, it's supplemental data from a subsequent hearing call, so I've added it as an additional attribute that can be populated when that data is available.

In both cases maybe I shouldn't have included them yet, as they're part of the adaptations which are still to be done. If it's cleaner I can remove them from both models for now and add them in a later PR.

Comment (archferns) [reply app/models/hearing_summary.rb]: That makes sense. I'm happy to have them in as it complements your serializer specs.

File: app/models/defendant.rb (hunk 2 above)
Comment (archferns) [reply app/models/defendant.rb]: makes sense. I'm happy to have them in as it complements your serializer specs.

**Pull request merged**